src/detection/cpu/cpu_linux.c

```diff
@@ -197,6 +197,86 @@ const char* ffDetectCPUImpl(const FFCPUOptions* options, FFCPUResult* cpu)
                 }
             }
 
+            // In Asahi Linux, reading /proc/device-tree/compatible gives
+            // information on the device model. It consists of 3 NUL terminated
+            // strings, the second of which gives the actual SoC model. But it
+            // is not the marketing name, i.e. for M2 there is "apple,t8112" in
+            // the compatible string.
+            //
+            // A full list of the SoC model names can be found here:
+            // https://github.com/AsahiLinux/docs/wiki/Codenames
+            #define DT_COMPAT_PATH "/proc/device-tree/compatible"
+            if (strcmp(cpu->vendor.chars, "Apple") == 0)
```

Comment:
ffStrbufEqualS(&cpu->vendor, "Apple")

Reply:
Where do you get the vendor name?

Reply:
The section above that runs `lscpu` and parses the  "Vendor ID" section correctly sets the vendor to "Apple". On my machine it looks like this:
```
Vendor ID:              Apple
  Model name:           Blizzard-M2
    Model:              0
```
I'll update to use the correct function.